inc/editor/blocks/toc.php

```diff
@@ -34,7 +34,10 @@ function get_headings_from_post_content( string $content ) : array {
 	// within PHP rendering code, even once the content is parsed as blocks.
 	// DOMDocument is the most reliable tool to locate the values we want.
 	$heading_block_doc = new DOMDocument();
-	$heading_block_doc->loadHTML( $content );
+	libxml_use_internal_errors( true ); // Silence warnings.
+	$content = preg_replace( '/<svg(.*?)<\/svg>/is', '', $content ); // Remove any SVG content.
```

Comment:
Why do we need to remove SVG?

Reply:
I saw many errors complaining for dealing with SVG:

```
PHP message: Warning: DOMDocument::loadHTML(): Tag svg invalid in Entity, line: 93 in /var/www/wp-content/themes/shiro/inc/editor/blocks/toc.php on line 37 [wikimediafoundation-org-develop.go-vip.co/about/annualreport/] [wp-content/themes/shiro/inc/editor/blocks/toc.php:37 DOMDocument->loadHTML(), wp-content/themes/shiro/inc/editor/blocks/toc.php:152 WMF\Editor\Blocks\TOC\get_headings_from_post_content(), wp-includes/class-wp-hook.php:312 WMF\Editor\Blocks\TOC\render_toc_block(), wp-includes/plugin.php:205 WP_Hook->apply_filters(), wp-includes/class-wp-block.php:293 apply_filters('render_block'), wp-includes/class-wp-block.php:244 WP_Block->render(), wp-includes/class-wp-block.php:244 WP_Block->render(), wp-includes/class-wp-block.php:244 WP_Block->render(), wp-includes/blocks.php:1133 WP_Block->render(), wp-includes/blocks.php:1171 render_block(), wp-includes/class-wp-hook.php:310 do_blocks(), wp-includes/plugin.php:205 WP_Hook->apply_filters(), wp-includes/post-template.php:256 apply_filters('the_content'), wp-content/themes/shiro/template-parts/content-page.php:16 the_content(), wp-includes/template.php:787 require('wp-content/themes/shiro/template-parts/content-page.php'), wp-includes/template.php:720 load_template(), wp-includes/general-template.php:206 locate_template(), wp-content/themes/shiro/page-block-editor.php:76 get_template_part(), wp-includes/template.php:787 require('wp-content/themes/shiro/page-block-editor.php'), wp-includes/template.php:720 load_template(), wp-includes/general-template.php:206 locate_template(), wp-content/themes/shiro/page.php:18 get_template_part(), wp-includes/template-loader.php:106 include('wp-content/themes/shiro/page.php'), wp-blog-header.php:19 require_once('wp-includes/template-loader.php'), index.php:17 require('wp-blog-header.php')]

```

Reply:
@MiguelAxcar DOMDocument doesn't properly detect HTML5 mode (despite that it's... (checks my watch) 2023!), so it will flag `nav`, `svg`, etc as invalid. We do want to support HTML5 elements in post content, and there's a chance that one of them may wrap a headline, so I recommend disabling the errors as you do now, but not altering the content string.